pidgin/app.py

```diff
@@ -0,0 +1,113 @@
+import json
+
+import flask
+import requests
+
+
+app = flask.Flask(__name__)
+
+
+# Endpoint to get core metadata as JSON from an object_id.
+@app.route('/json/<path:object_id>') # 'path' allows the use of '/' in the id
+def get_json_metadata(object_id):
+    try:
+        metadata = get_metadata_dict(object_id)
+        return translate_dict_to_json(metadata)
+    except Exception as e:
+        return str(e), 300
+
+# Endpoint to get core metadata as BibTeX from an object_id.
+@app.route('/bibtex/<path:object_id>') # 'path' allows the use of '/' in the id
+def get_bibtex_metadata(object_id):
+    metadata = get_metadata_dict(object_id)
+    return translate_dict_to_bibtex(metadata)
+
+
+# Create a dictionary containing the metadata for a given object_id.
+def get_metadata_dict(object_id):
+    response = request_metadata(object_id) # query to peregrine
+    data = response.get_json()
+    return flatten_dict(data) # translate the response to a dictionary
+
+
+# Translate a dictionary to a JSON string.
+def translate_dict_to_json(d):
+    json_str = json.dumps(d)
```

Comment:
This function `translate_dict_to_json` seems to be just duplicating the built-in `json.dumps`, so I'd suggest removing this function (and then you don't have to worry about testing it either)